8.8.2 Release notes (#3472)
elastic/security-docs: docs/release-notes/8.8.asciidoc

[[release-notes-header-8.8.0]]
== 8.8

[discrete]
[[release-notes-8.8.2]]
=== 8.8.2

[discrete]
[[breaking-changes-8.8.2]]
==== Breaking changes

//tag::breaking-changes[]
// NOTE: The breaking-changes tagged regions are reused in the Elastic Installation and Upgrade Guide. The pull attribute is defined within this snippet so it properly resolves in the output.
// THIS ALSO MEANS IF YOU USE LINKS HERE, THEY SHOULD BE FULL URLS WITH NO ATTRIBUTES

:pull: https://github.com/elastic/kibana/pull/

There are no breaking changes in 8.8.2.

//end::breaking-changes[]

[discrete]

> [[enhancements-8.7.1]]

Author: ```suggestion
[[enhancements-8.8.2]]
```

+
NOTE: If you've already encountered this issue and want to find erroneously duplicated exceptions, use the queries provided below.

** **Query for finding exception documents that were duplicated from only specifying the `item_id`:**
+
[source,kibana]
----------------------------------
// Retrieve exception documents grouped by `item_id`. 
// Each bucket contains all duplicates of that document
GET .kibana*/_search
{
  "query": {
    "bool": {
      "filter": [
        {
          "match": {
            "type": "exception-list"
          }
        }
      ]
    }
  },
  "aggs": {
    "item_id_duplicates": {
      "terms": {
        "field": "exception-list.item_id",
        "min_doc_count": 2
      },
      "aggs": {
        "ids": {
          "top_hits": {
            "size": 100, // increase this if you may have more duplicates
            "_source": false
          }
        }
      }
    }
  },
  "size": 0
}
----------------------------------

** **Query for finding exception documents that were duplicated and have lost their `item_id` because their `id` was used to update them:**

> ----------------------------------

Reviewer: ```suggestion
[source,kibana]
----------------------------------
```

There are no breaking changes in 8.8.2.

//end::breaking-changes[]

[discrete]
[[enhancements-8.8.2]]
==== Enhancements
There are no user-facing changes in 8.8.2.

[discrete]
[[bug-fixes-8.8.2]]
==== Bug fixes 
* Fixes a bug that affected links to results in Kibana that were generated by actions on rules from non-default {kib} spaces ({pull}159966[#159966]).
* Fixes a bug that prevented users from saving a Timeline after adding a `number` field to it ({pull}159723[#159723]).
* Fixes a bug that caused error messages to wrongfully display if users selected uninstalled {ml} jobs while creating a {ml} rule ({pull}159316[#159316]).
* Removes hover actions from table within the Detection & Response dashboard when the alert count is zero ({pull}158902[#158902]).
* Fixes bugs in the Anomalies table that left {ml} jobs greyed out after they were installed ({pull}158821[#158821]) and that stopped some job counts from appearing ({pull}158739[#158739]).
* Fixes a bug that caused the **Add exceptions flyout** to load indefinitely and display an out of memory error when a rule had a large number of unmapped fields in multiple indices ({pull}159216[#159216]).

> * Allows actions to be taken on fields with multiple values ({pull}158060[#158060]).

Reviewer: ```suggestion
* Allows actions to be taken on fields with multiple values ({pull}158060[#158060]).
* Fixes a bug that caused the Alerts page to query unnecessary indices ({pull}157286[#157286]).
* Fixes a bug in 8.8.0 and 8.8.1 that broke the moustache syntax for variables in rule actions.
```

* Fixes a bug that caused the Alerts page to query unnecessary indices ({pull}157286[#157286]).

> * Fixes a bug that broke the mustache syntax for variables in rule actions.

Reviewer: ```suggestion
* Fixes a bug that broke the mustache syntax for variables in rule actions ({pull}160446[#160446]).
```

* Fixes a bug that caused exception items to be erroneously duplicated if you modified an exception item using the <<exceptions-api-update-item,update exception item>> API and _only_ specified its `item_id` ({pull}159223[#159223]).
+
NOTE: If you've already encountered this issue and want to find erroneously duplicated exceptions, use the queries provided below.

** **Query for finding exception documents that were duplicated from only specifying the `item_id`:**
+
[source,kibana]
----------------------------------
// Retrieve exception documents grouped by `item_id`. 
// Each bucket contains all duplicates of that document
GET .kibana*/_search
{
  "query": {
    "bool": {
      "filter": [
        {
          "match": {
            "type": "exception-list"
          }
        }
      ]
    }
  },
  "aggs": {
    "item_id_duplicates": {
      "terms": {
        "field": "exception-list.item_id",
        "min_doc_count": 2
      },
      "aggs": {
        "ids": {
          "top_hits": {
            "size": 100, // increase this if you may have more duplicates
            "_source": false
          }
        }
      }
    }
  },
  "size": 0
}

> ```

Reviewer: ```suggestion
```

** **Query for finding exception documents that were duplicated and have lost their `item_id` because their `id` was used to update them:**

> 

Reviewer: ```suggestion
+
```

:pull: https://github.com/elastic/kibana/pull/

There are no breaking changes in 8.8.2.

//end::breaking-changes[]

[discrete]
[[enhancements-8.8.2]]
==== Enhancements
There are no user-facing changes in 8.8.2.

[discrete]
[[bug-fixes-8.8.2]]
==== Bug fixes 

> * Fixes a bug that affected links generated by actions added to rules that were created in non-default {kib} spaces ({pull}159966[#159966]).

Reviewer 2: ```suggestion
* Fixes a bug that affected links to results in Kibana, generated by actions added to rules that were created in non-default {kib} spaces ({pull}159966[#159966]).
```

* Fixes a bug that prevented users from saving a Timeline after adding a `number` field to it ({pull}159723[#159723]).
* Fixes a bug that caused error messages to wrongfully display if users selected uninstalled {ml} jobs while creating a {ml} rule ({pull}159316[#159316]).
* Removes hover actions from table within the Detection & Response dashboard when the alert count is zero ({pull}158902[#158902]).

> * Fixes bugs in the Anomalies table that left {ml} jobs greyed out after they were installed ({pull}158821[#158821]) and that stopped some job counts from being shown ({pull}158739[#158739]).

Reviewer: ```suggestion
* Fixes bugs in the Anomalies table that left {ml} jobs greyed out after they were installed ({pull}158821[#158821]) and that stopped some job counts from appearing ({pull}158739[#158739]).
```

* Fixes a bug that caused the **Add exceptions flyout** to load indefinitely and display an out of memory error when a rule had a large number of unmapped fields in multiple indices ({pull}159216[#159216]).
* Allows actions to be taken on fields with multiple values ({pull}158060[#158060]).
* Fixes a bug that caused exception items to be erroneously duplicated if you modified an exception item using the <<exceptions-api-update-item,update exception item>> API and _only_ specified its `item_id` ({pull}159223[#159223]).

> +

Reviewer: ```suggestion
+ * Fixes a bug that caused the Alerts page to query unnecessary indices.
```

> + * Fixes a bug that caused the Alerts page to query unnecessary indices ({pull}157286[#157286]).

Reviewer: ```suggestion
```

If you've already encountered this issue and want to find erroneously duplicated exceptions, use the queries provided below.

> ** **Query for finding exception documents that were duplicated from only specifying the `item_id`:**

Reviewer 3: @rylnd can you confirm this looks good?

Reviewer 4: LGTM!

** **Query for finding exception documents that were duplicated from only specifying the `item_id`:**
+
[source,kibana]
----------------------------------

> ```

Reviewer: ```suggestion
```

// Retrieve exception documents grouped by `item_id`. 
// Each bucket contains all duplicates of that document
GET .kibana*/_search
{
  "query": {
    "bool": {
      "filter": [
        {
          "match": {
            "type": "exception-list"
          }
        }
      ]
    }
  },
  "aggs": {
    "item_id_duplicates": {
      "terms": {
        "field": "exception-list.item_id",
        "min_doc_count": 2
      },
      "aggs": {
        "ids": {
          "top_hits": {
            "size": 100, // increase this if you may have more duplicates
            "_source": false
          }
        }
      }
    }
  },
  "size": 0
}
```

> =======

Reviewer: ```suggestion
----------------------------------
```

:pull: https://github.com/elastic/kibana/pull/

There are no breaking changes in 8.8.2.

//end::breaking-changes[]

[discrete]
[[enhancements-8.8.2]]
==== Enhancements
There are no user-facing changes in 8.8.2.

[discrete]
[[bug-fixes-8.8.2]]
==== Bug fixes 

> * Fixes a bug that affected links to results in Kibana that were generated by actions on rules from non-default {kib} spaces ({pull}159966[#159966]).

Reviewer 5: ```suggestion
* Fixes a bug that affected links to {kib} results generated by actions on rules from non-default {kib} spaces ({pull}159966[#159966]).
```

> * Fixes a bug that prevented users from saving a Timeline after adding a `number` field to it ({pull}159723[#159723]).

Reviewer 5: ```suggestion
* Fixes a bug that prevented users from saving a Timeline after adding a `number` field ({pull}159723[#159723]).
```

* Fixes a bug that caused error messages to wrongfully display if users selected uninstalled {ml} jobs while creating a {ml} rule ({pull}159316[#159316]).

> * Removes hover actions from table within the Detection & Response dashboard when the alert count is zero ({pull}158902[#158902]).

Reviewer 5: Assuming this applies to all the tables in the D&R dashboard. 
```suggestion
* Removes hover actions from tables within the Detection & Response dashboard when the alert count is zero ({pull}158902[#158902]).
```

* Fixes bugs in the Anomalies table that left {ml} jobs greyed out after they were installed ({pull}158821[#158821]) and that stopped some job counts from appearing ({pull}158739[#158739]).
* Fixes a bug that caused the **Add exceptions flyout** to load indefinitely and display an out of memory error when a rule had a large number of unmapped fields in multiple indices ({pull}159216[#159216]).
* Allows actions to be taken on fields with multiple values ({pull}158060[#158060]).
* Fixes a bug that caused the Alerts page to query unnecessary indices ({pull}157286[#157286]).

> * Fixes a bug in 8.8.0 and 8.8.1 that broke the moustache syntax for variables in rule actions.

Reviewer 5: ```suggestion
* Fixes a bug that broke the mustache syntax for variables in rule actions.
```

* Fixes a bug that caused exception items to be erroneously duplicated if you modified an exception item using the <<exceptions-api-update-item,update exception item>> API and _only_ specified its `item_id` ({pull}159223[#159223]).

If you've already encountered this issue and want to find erroneously duplicated exceptions, use the queries provided below.

** **Query for finding exception documents that were duplicated from only specifying the `item_id`:**
+

> =======

Reviewer 6: I tested a few things locally and `[source,kibana]` works well for me. The block definitely needs to be dashes (`----`) not equals signs, and the backticks (```) aren't needed. So basically just like @jmikell821's example above.

```

> // Retrieve exception documents grouped by `item_id`. 

Reviewer 5: Same comment as above about the code blocks.

// Each bucket contains all duplicates of that document
GET .kibana*/_search
{
  "query": {
    "bool": {
      "filter": [
        {
          "match": {
            "type": "exception-list"
          }
        }
      ]
    }
  },
  "aggs": {
    "item_id_duplicates": {
      "terms": {
        "field": "exception-list.item_id",
        "min_doc_count": 2
      },
      "aggs": {
        "ids": {
          "top_hits": {
            "size": 100, // increase this if you may have more duplicates
            "_source": false
          }
        }
      }
    }
  },
  "size": 0
}
```
=======

** **Query for finding exception documents that were duplicated and have lost their `item_id` because their `id` was used to update them:**
+
=======

> ```

Reviewer 5: Same comment as above about the code blocks.

// Each item returned lost its `item_id`, which is expected to be present and unique
GET .kibana*/_search
{
  "query": {
    "bool": {
      "filter": [
        {
          "term": {
            "exception-list.list_type": "item"
          }
        }
      ],
      "must_not": [
        {
          "exists": {
            "field": "exception-list.item_id"
          }
        }
      ]
    }
  }
}

> ```

Reviewer: ```suggestion
```

> =======

Reviewer: ```suggestion
----------------------------------
```

[discrete]
[[bug-fixes-8.8.2]]
==== Bug fixes 
* Fixes a bug that affected links to {kib} results generated by actions on rules from non-default {kib} spaces ({pull}159966[#159966]).
* Fixes a bug that prevented users from saving a Timeline after adding a `number` field ({pull}159723[#159723]).
* Fixes a bug that caused error messages to wrongfully display if users selected uninstalled {ml} jobs while creating a {ml} rule ({pull}159316[#159316]).
* Removes hover actions from tables within the Detection & Response dashboard when the alert count is zero ({pull}158902[#158902]).
* Fixes bugs in the Anomalies table that left {ml} jobs greyed out after they were installed ({pull}158821[#158821]) and that stopped some job counts from appearing ({pull}158739[#158739]).
* Fixes a bug that caused the **Add exceptions flyout** to load indefinitely and display an out of memory error when a rule had a large number of unmapped fields in multiple indices ({pull}159216[#159216]).
* Allows actions to be taken on fields with multiple values ({pull}158060[#158060]).
* Fixes a bug that caused the Alerts page to query unnecessary indices ({pull}157286[#157286]).

> * Fixes a bug that broke the mustache syntax for variables in rule actions.

Reviewer 6: Missing PR link?

* Fixes a bug that caused exception items to be erroneously duplicated if you modified an exception item using the <<exceptions-api-update-item,update exception item>> API and _only_ specified its `item_id` ({pull}159223[#159223]).
+
NOTE: If you've already encountered this issue and want to find erroneously duplicated exceptions, use the queries provided below.

** **Query for finding exception documents that were duplicated from only specifying the `item_id`:**
+
[source,kibana]

> =======

Reviewer: ```suggestion
----------------------------------
```

```
// Retrieve exception documents grouped by `item_id`. 
// Each bucket contains all duplicates of that document
GET .kibana*/_search
{
  "query": {
    "bool": {
      "filter": [
        {
          "match": {
            "type": "exception-list"
          }
        }
      ]
    }
  },
  "aggs": {
    "item_id_duplicates": {
      "terms": {
        "field": "exception-list.item_id",
        "min_doc_count": 2
      },
      "aggs": {
        "ids": {
          "top_hits": {
            "size": 100, // increase this if you may have more duplicates
            "_source": false
          }
        }
      }
    }
  },
  "size": 0
}
```
=======

** **Query for finding exception documents that were duplicated and have lost their `item_id` because their `id` was used to update them:**

> +
> =======
> ```

Reviewer: ```suggestion

----------------------------------
```

[discrete]
[[bug-fixes-8.8.1]]
==== Bug fixes 

> * Fixes bugs in the Anomalies table that left {ml} jobs greyed out after they were installed ({pull}158821[#158821]), and that stopped some job counts from being shown ({pull}158739[#158739]).

Author: @machadoum would you mind confirming that this was only fixed in 8.8.2 and _not_ 8.8.1? I removed this summary after noticing [this comment](https://github.com/elastic/kibana/pull/158739#issuecomment-1583068988) about the fix not making it into the latest BC for 8.8.1.

Reviewer 7: I confirm. It is only fixed on 8.8.2.

* Fixes a bug that caused all field types in the **Fields** browser to appear as `unknown` ({pull}158594[#158594]).
* Fixes a bug that caused the **Add rule exception** flyout to load indefinitely when index fields couldn't be retrieved ({pull}158371[#158371]).

> * Allows actions to be taken on fields with multiple values ({pull}158060[#158060]).

Author: @machadoum would you mind confirming that this was only fixed in 8.8.2 and _not_ 8.8.1? I removed this summary after noticing [this comment](https://github.com/elastic/kibana/pull/158060#issuecomment-1583069201) about the fix not making it into the latest BC for 8.8.1.

Reviewer 7: Hey! You can trust the comment. It didn't make it for 8.8.1.
